### DISCUSSION TITLE
Not all image attachments are embedded

### DISCUSSION THREAD

**Discussion by xygn** [srid/emanote#97] [Q&A] [answered]
Hi!

I am using emanote to host my markdown notes on the web. However, I've noticed that not all images linked within the file are embedded as an image. Rather they are displayed as an attachment, that on clicking leads you to the static/ source.

Here is an [example note:](https://n.xygn.us/C-Notes/Core/Pathology/Systemic/2.%20Inflammation)
[Source](https://gist.githubusercontent.com/xygn/10b6e9d75ef9368b40663d7b97bf03c9/raw/088195dc53c943070fb66c75b7d4302cebff31d9/Inflammation.md)

```
*Funcio lassiea* / Loss of function - Virchow
![[Pasted image 20210624232124.png]]
```

In the webpage generated you can see it as:
![image](https://user-images.githubusercontent.com/63365609/124117968-646cdf80-da8e-11eb-928e-98726ca8cbb2.png)

I am unable to figure out what is causing this, as a few pictures are embedded and some aren't.

Is there anything I'm doing wrong?

[Debug Log](https://gist.githubusercontent.com/xygn/8292465ddce010a2246ccd5fe34806c8/raw/e882d4b222a1473a1741b1011b9400bbeeaa0e99/emanotedebug.log)

Regards,
xygn

**Comment by srid** [comment, marked as answer]
From https://note.ema.srid.ca/demo/embed

> Note that the embed wiki-link syntax must appear on a paragraph of its own, with no other text added next to it.

In code,

https://github.com/srid/emanote/blob/8b85459c000cd9100ebdd708ec47dfd411c72450/src/Emanote/Pandoc/Filter/Embed.hs#L31

---

I see a number of images in that link inside list elements. Now list elements can contain paragraphs. So what if you make it like the following?

```markdown
1. First element
2. Second element
    
    ![[Some Image]]
3. Third element
```

---

The reason for this constraint is mentioned in the footnote of the link further above,

> This constraint is necessary to ensure that the HTML generated remains valid. Embedded content use block elements, which cannot be embedded inside inline nodes.

This applies to embedding notes (because that would create a `<div>` a block element), however embedding images involves creating a `<img>` element only, which is *not* a block element - so ideally we should allow that to happen - such that your notebook continues to work in its current state without requiring hacks like the above. I've filed https://github.com/srid/emanote/issues/98 for that.

**Reply by xygn** [reply]
Ah I see. Yes I corrected it to your approach, it's working fine now. Thank you! 

I also wanted to point out that the bullet styling is a bit off sometimes. The text shows below the bullet point instead of beside it. 

![image](https://user-images.githubusercontent.com/63365609/124153824-7a8c9700-dab2-11eb-9aee-e12d764bce82.png)


The same linked example can be used for reference.

**Reply by srid** [reply]
@xygn Could you file a bug for this? Include Markdown and the generated HTML.

**Reply by srid** [reply]
Oh n/m. I see you included the source in your original post.

**Reply by srid** [reply]
Some of that is due to use of `</br>` and unnecessary empty line in the midst of lists, eg:

```markdown
- *-itis* denotes inflammation </br>

- Types
	- Purulent infl. 
	- Serous infl.
```

**Reply by srid** [reply]
Filed a bug, https://github.com/srid/emanote/issues/99

**Reply by xygn** [reply]
> Some of that is due to use of `</br>` and unnecessary empty line in the midst of lists, eg:
> 
> ```gfm
> - *-itis* denotes inflammation </br>
> 
> - Types
> 	- Purulent infl. 
> 	- Serous infl.
> ```

I was testing out if the empty lines or linebreaks would help, but even when it's just a normal list without unnecessary lines, it has the same result. 



> Filed a bug, #99

I have attached the generated HTML [#99](https://github.com/srid/emanote/issues/99#issuecomment-872384128)

**Reply by srid** [reply]
Is this now resolved for you, @xygn ?
